app/code/community/Bolt/Boltpay/controllers/ShippingController.php

```diff
@@ -132,11 +134,13 @@ public function indexAction()
             // Check session cache for estimate.  If the shipping city or postcode, and the country code match,
             // then use the cached version.  Otherwise, we have to do another calculation
             ////////////////////////////////////////////////////////////////////////////////////////
-            $cached_address = unserialize(Mage::app()->getCache()->load("quote_location_".$quote->getId()));
+            $cacheKeyLocation = $quote->getId() . '_' . $address_data['country_id'];
+            $prefetchCacheKey = $this->generateCacheKey($quote, $address_data);
```

Comment:
I think we need to see if the $prefetchCacheKey actually exists in the cache. I just tested on my local and it doesn't, so the response returned is empty and no shipping rates are returned at all. I think we might be over complicating this. I would suggest that we create an array that contains all of the fields that we care about (e.g. quote id, street address, city, state, postal code, country, grand total, skus, etc), populate it in generateCacheKey(), and store that as a single checksum rather than keeping two cache keys. If it exists then we can know whether or not the quote is exactly the same or not, rather than just checking that the city, postcode, and country are the same. If so we use that checksum to return the response. If not, we create a new one.